fixham/internal/registry/registry.go

```diff
@@ -0,0 +1,102 @@
+package registry
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+const (
+	QuayRegistry   = "quay.io"
+	DockerRegistry = "docker.io"
+)
+
+// Image represents a container image.
+type Image string
+
+// Repository returns the repository part of the image.
+func (d Image) Repository() string {
+	parts := strings.Split(string(d), ":")
+	parts = strings.Split(parts[0], "/")
+	return parts[0]
+}
+
+// Tag returns the tag part of the image.
+func (d Image) Tag() string {
+	parts := strings.Split(string(d), ":")
+	if len(parts) > 1 {
+		return parts[1]
+	}
+	return "latest"
+}
+
+// Registry represents a container registry.
+type Registry interface {
+	URL() string
+	TokenURL(repository string) string
+	ManifestURL(img Image) string
+}
+
+// GetRegistry returns a Registry based on the registry string.
+func GetRegistry(registry string) Registry {
+	switch registry {
+	case QuayRegistry:
+		return &Quay{}
+	case DockerRegistry:
+		return &Docker{}
+	default:
+		return &Docker{}
+	}
+}
+
+// getBearerToken retrieves a bearer token to use for the image.
+func getBearerToken(registry Registry, img Image) (string, error) {
+	tokenURL := registry.TokenURL(img.Repository())
+	req, err := http.NewRequest(http.MethodGet, tokenURL, nil)
+	if err != nil {
+		return "", err
+	}
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("failed to get bearer token: %s", resp.Status)
+	}
+	var tokenResp struct {
```

Comment:
This is pretty unusual.

I think I'd opt for defining the struct type and variable separately just for more standard looking code. 

```go
type tokenResp struct {
}
var r tokenResp
```

or just using a map[string]interface{]

```go
resp := map[string]interface{}{}
if err := json.NewDecoder(resp.Body).Decode(&resp); err != nil {
  return "", err
}
return resp["token"].(string), nil
```